databuilder/databuilder/models/type_metadata.py

```diff
@@ -29,21 +33,43 @@ def __init__(self,
                  name: str,
                  parent: Union[ColumnMetadata, 'TypeMetadata'],
                  type_str: str,
-                 description: Optional[str] = None,
                  sort_order: Optional[int] = None) -> None:
         self.name = name
         self.parent = parent
         self.type_str = type_str
-        self.description = DescriptionMetadata.create_description_metadata(
-            source=None,
-            text=description
-        )
         # Sort order among TypeMetadata objects with the same parent
         self.sort_order = sort_order
 
+        self._description: Optional[DescriptionMetadata] = None
+        self._badges: Optional[List[Badge]] = None
+
         self._node_iter = self.create_node_iterator()
         self._relation_iter = self.create_relation_iterator()
 
+    def get_description(self) -> Optional[DescriptionMetadata]:
+        return self._description
+
+    def set_description(self, description: str) -> None:
+        if isinstance(self.parent, ColumnMetadata) or isinstance(self.parent, ArrayTypeMetadata):
+            LOGGER.warning(f"Frontend does not currently support setting descriptions for type metadata "
+                           f"objects with a {type(self.parent)} parent")
```

Comment:
hardcode which ones, and give a reason. I think it might be better actually to split these into two `if`s .